packages/checkout-extension/src/ExtensionService.ts

```diff
@@ -0,0 +1,111 @@
+import {
+    CheckoutService,
+    Extension,
+    ExtensionCommandType,
+    ExtensionRegion,
+} from '@bigcommerce/checkout-sdk';
+
+export class ExtensionService {
+    private _handlerRemovers: { [extensionId: string]: Array<() => void> } = {};
+
+    constructor(
+        private _checkoutService: CheckoutService,
+        private _setIsShowLoadingIndicator: (show: boolean) => void,
+    ) {}
+
+    async loadExtensions(): Promise<void> {
+        await this._checkoutService.loadExtensions();
+    }
+
+    async renderExtension(container: string, region: ExtensionRegion): Promise<void> {
+        const extension = this._getExtensionByRegion(region);
+
+        if (!extension) {
+            return;
+        }
+
+        await this._checkoutService.renderExtension(container, region);
+
+        if (!this._handlerRemovers[extension.id]) {
+            this._handlerRemovers[extension.id] = [];
+        }
+
+        this._handlerRemovers[extension.id].push(
+            this._checkoutService.handleExtensionCommand(
+                extension.id,
+                ExtensionCommandType.ReloadCheckout,
+                () => {
+                    void this._checkoutService.loadCheckout(
+                        this._checkoutService.getState().data.getCheckout()?.id,
+                    );
+                },
+            ),
+        );
+
+        this._handlerRemovers[extension.id].push(
+            this._checkoutService.handleExtensionCommand(
+                extension.id,
+                ExtensionCommandType.SetIframeStyle,
+                (data) => {
+                    if (data.type === ExtensionCommandType.SetIframeStyle) {
```

Comment:
I think we can make it so that `handleExtensionCommand` can map `ExtensionCommandType.SetIframeStyle` to a different handler signature. For example, something like this: https://github.com/bigcommerce/checkout-sdk-js/blob/38555c99fe63c6dfccf767c1f9c3ff8962b4cfee/packages/core/src/common/iframe/iframe-event-listener.ts#L41-L44. That way you don't need to do this check on the client side.

Reply:
Thanks for pointing out the mapping.
I have added this as a new task in the follow-up SDK ticket.